python/neutron-understack/neutron_understack/neutron_understack_mech.py

```diff
@@ -202,15 +200,54 @@ def update_port_precommit(self, context):
     def update_port_postcommit(self, context):
         log_call("update_port_postcommit", context)
 
+        if context.current["binding:vif_type"] == portbindings.VIF_TYPE_OTHER:
+            self._move_to_network(context)
+
     def delete_port_precommit(self, context):
         log_call("delete_port_precommit", context)
 
     def delete_port_postcommit(self, context):
         log_call("delete_port_postcommit", context)
 
     def bind_port(self, context):
-        self._move_to_network(context)
         log_call("bind_port", context)
+        for segment in context.network.network_segments:
+            if self.check_segment(segment):
+                context.set_binding(
+                    segment[api.ID],
+                    portbindings.VIF_TYPE_OTHER,
+                    {},
+                    status=p_const.PORT_STATUS_ACTIVE,
+                )
+                LOG.debug(f"Bound segment: {segment}")
+                return
+            else:
+                LOG.debug(
+                    "Refusing to bind port for segment ID %(id)s, "
+                    "segment %(seg)s, phys net %(physnet)s, and "
+                    "network type %(nettype)s",
+                    {
+                        "id": segment[api.ID],
+                        "seg": segment[api.SEGMENTATION_ID],
+                        "physnet": segment[api.PHYSICAL_NETWORK],
+                        "nettype": segment[api.NETWORK_TYPE],
+                    },
+                )
+
+    def check_segment(self, segment):
+        """Verify a segment is valid for the Understack MechanismDriver.
+
+        Verify the requested segment is supported by Understack and return True or
+        False to indicate this to callers.
+        """
+        network_type = segment[api.NETWORK_TYPE]
+        return network_type in [
+            p_const.TYPE_LOCAL,
+            p_const.TYPE_GRE,
+            p_const.TYPE_VXLAN,
+            p_const.TYPE_VLAN,
+            p_const.TYPE_FLAT,
```

Comment:
I think ultimately we'll want to cull this list down.

Reply:
Agreed, we probably can get away with just a TYPE_FLAT